clang/lib/CodeGen/Targets/PPC.cpp

```diff
@@ -337,12 +350,77 @@ CharUnits PPC32_SVR4_ABIInfo::getParamTypeAlignment(QualType Ty) const {
   return CharUnits::fromQuantity(4);
 }
 
+ABIArgInfo PPC32_SVR4_ABIInfo::handleComplex(QualType Ty,
+                                             uint64_t &TypeSize) const {
+
```

Comment:
it look the TypeSize only has 32bit and 64bit. can you change the `uint64_t &TypeSize` to `bool is64BitTypeSize`

and change following code as 

```
llvm::Type *ElemTy =is64BitTypeSize? llvm::Type::getInt64Ty(getVMContext()): llvm::Type::getInt32Ty(getVMContext());

unsigned SizeRegs = is64BitTypeSize ? 1:2;
```